modules/rostests/apitests/shell32/ShellExecuteW.cpp

```diff
@@ -2,178 +2,114 @@
  * PROJECT:         ReactOS api tests
  * LICENSE:         GPLv2+ - See COPYING in the top level directory
  * PURPOSE:         Testing ShellExecuteW
- * PROGRAMMER:      Doug Lyons <douglyons@douglyons.com>
+ * PROGRAMMERS:     Doug Lyons <douglyons@douglyons.com>
+ *                  Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
  */
 
 #include "shelltest.h"
 #include <stdio.h>
 #include <winbase.h>
 
-static
-VOID
-TestShellExecuteW()
+#define WAIT_SLEEP 700
+
+// ShellExecuteW(handle, "open", <path_to_executable>, <parameters>, NULL, SW_SHOWNORMAL);
+
+START_TEST(ShellExecuteW)
 {
+    INT ret;
     HINSTANCE hInstance;
-    INT retval;
     HWND hWnd;
-    const WCHAR *Name = L"ShellExecuteW";
-    WCHAR WindowsDirectory[MAX_PATH];
-    BOOL IsReactOS;
-
-    /* Check if we are running under ReactOS from the SystemRoot directory */
-    if(!GetWindowsDirectoryW(WindowsDirectory, MAX_PATH))
-        printf("GetWindowsDirectoryW failed\n");
-
-    IsReactOS = !_wcsnicmp(&WindowsDirectory[3], L"reactos", 7);
-
-    printf("OSVendor %s ReactOS.\n", IsReactOS ? "is" : "is not");
+    WCHAR WinDir[MAX_PATH], SysDir[MAX_PATH];
 
-//  ShellExecuteW(handle, "open", <fully_qualified_path_to_executable>, <parameters>, NULL, SW_SHOWNORMAL);
+    if (!GetWindowsDirectoryW(WinDir, MAX_PATH))
```

Comment:
Instead of hardcoding the MAX_PATH here and in the other function call below, you can use either `ARRAYSIZE(WinDir)` or `__countof(WinDir)`, whichever compiles.

Reply:
OK.